### PR TITLE
[dynamo] Replace set by OrderedSet

### PR DESCRIPTION
**Pull request by rec**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #139645
* #138454

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

File: torch/_dynamo/backends/tvm.py [hunk 1]
```diff
@@ -149,7 +150,7 @@ def to_tvm_tensor(torch_tensor):
     def exec_tvm(*i_args):
         args = [a.contiguous() for a in i_args]
         shape_info, _ = m.get_input_info()
-        active_inputs = {name for name, _ in shape_info.items()}
+        active_inputs = OrderedSet([name for name, _ in shape_info.items()])
```
Comment (Skylion007) [inline torch/_dynamo/backends/tvm.py:153]:
```suggestion
        active_inputs = OrderedSet((name for name, _ in shape_info.items()))
``` 
Does it really need a temp list?

Comment (rec) [reply torch/_dynamo/backends/tvm.py:153]: Absolutely it does not. :-D

Fixing all of these right now.  (these edits were performed automatically, you see...)

Comment (rec) [reply torch/_dynamo/backends/tvm.py]: @Skylion007: done.

Note that this is a draft - it's really not clear that Dynamo suffers from the same issues that Inductor has with plain old `set` but given this linter/fixer tool, I might as well try it out!

Comment (Skylion007) [reply torch/_dynamo/backends/tvm.py]: Cool, just wanted to make sure I didn't have to update OrderedSet's typing.

**Comment by rec** [comment]
@anijain2305 @Skylion007

I'm not actually convinced that this pull request is worth doing - unlike inductor, dynamo hasn't (yet?) run into any issues with the indeterministic iteration order of classic `set`.

However, I have a [inter](https://github.com/pytorch/pytorch/pull/138454) that (mostly) automates the fix, and I needed to try it on something other than inductor, so here it is.

**Comment by rec** [comment]
Please see [this comment](https://github.com/pytorch/pytorch/pull/138466#issuecomment-2462344146) about the replacement of small sets by `tuple` instead of `OrderedSet`.

**Review by williamwen42 — COMMENTED** [review]
I'm not sure I see too much value from this since most of dynamo is not dependent on set ordering. This change has quite a large footprint, and we can also question how we will actually enforce using ordered sets throughout dynamo going forward. For example, most of the bytecode manipulation functions don't require set ordering.

We should only use ordered sets where we have determined them to be necessary. Most places that require ordering (e.g. in guards or output graphs) are already checked for order anyway, and ordering is implemented through other means (lists, dicts, etc.)

**Pull request closed without merging**